README.rst

```diff
@@ -179,6 +182,138 @@ Additionally Supported Parameters
+
+.. note::
+    A notification type configuration should contain atleast one of ``description`` or ``message_template``
+    settings. If both of them are present, ``description`` is given preference over ``message_template``. 
+
+Defining ``message_template``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can either extend default message template or write your own from scratch. An example to extend default
+message template is shown below.
+
+.. code-block:: jinja2
+
+    # In templates/openwisp_notifications/your_message_template.md
+    {% extends 'openwisp_notifications/message.md' %}
+    {% block body %}
+        {{ opts.target }} has malfunctioned. 
+    {% endblock body %}
+
+.. note::
+    You can access all attributes of the notification using ``opts`` variables in your message template as shown above.
+
+Registering/Unregistering Notification Types
+--------------------------------------------
+**OpenWISP Notifications** provides registering and unregistering notifications through utility functions
+``openwisp_notifications.types.register_notification_type`` and ``openwisp_notifications.types.unregister_notification_type``. Using
+these functions you can register or unregister notification types from anywhere in your code. 
+
+register_notification_type
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+This function is used to register a new notification type from anywhere in your code.
+Syntax:
+
+.. code-block:: python
+
+    register_notification_type(type_name, type_config)
+
++---------------+--------------------------------------------------------------+
+|   Parameter   |                     Description                              |
++---------------+--------------------------------------------------------------+
+|   type_name   | A ``str`` defining name of the notification type.            |
++---------------+--------------------------------------------------------------+
+|  type_config  | A ``dict`` defining configuration of the notification type.  |
++---------------+--------------------------------------------------------------+
+
+An example usage has been shown below.
+
+.. code-block:: python
+
+    from openwisp_notifications.types.register_notification_type
+
+    # Define configuration of your notification type
+    custom_type = {
+            'level': 'info',
+            'verb': 'added',
+            'verbose_name': 'device added',
+            'description': '{opts.target} was {opts.verb} at {opts.timestamp}',
+            'email_subject' : '[{site.name}] A device has been added'
+    }
+
+    # Register your custom notification type
+    register_notification_type('custom_type', custom_type)
+
+.. note::
+
+    It will raise ``ImproperlyConfigured`` exception if a notification type is already registered
+    with same name(not to be confused with verbose_name).
+
+.. note::
+
+    You can use ``site`` and ``opts`` variables while defining ``description`` and ``email_subject``
```

Comment:
`opts` sounds a bit misleading, because it's usually referred for model options, I think `instance`, `obj` or `notification` would be more suited.